jax/experimental/jax2tf/jax2tf.py

```diff
@@ -285,6 +296,26 @@ def convert(fun_jax: Callable,
   if experimental_native_lowering and not enable_xla:
     raise ValueError(
         "experimental_native_lowering is not supported with enable_xla=False")
+
+  if experimental_native_lowering and not experimental_native_lowering_platforms:
```

Comment:
One idea: Since you already know which backend it is here since the user is telling you, why not encode that as an in_sharding (maybe a SingleDeviceSharding or any other sharding) and then pass that around?

That way this stays internal and the threading of lowering_platform_override is not needed in pjit and pxla.py?

Reply:
I think that passing the platform in in_shardings may ultimately be the right solution. It would be quite involved, I think, because I would have to use some fake devices in the shardings (we want to be able to lower even for backends that we cannot initialize). The real reason I did not do it like that for now is that I could not follow the various places in the code where the shardings are processes. Afaik, these are in part due to supporting both jax.Array and jit==pjit and device or backend specified in the API. I would have to replicate some of these same heuristics to create shardings at the jax2tf boundary, because I have to handle the same cases that jit and pjit handle now (with device, backend, with or without shardings, ...).

Reply:
I should point out that this is just the first step towards multi-platform lowering. There we will lower on JAX function to a HLO module that includes the code for multiple platforms. The idea is that for the vast majority of the JAX primitives the lowering is platform-independent, but for those few cases where there is a different we will introduce a conditional in the code to check the value of a platform identifier that will be provided when the code is compiled and executed. In order to support multi-platform lowering, the code in this PR will be changed to replace `lowering_platform_override` with `lowering_platforms_override` (not the plural).

If we had a single platform to override, then we could somehow squirrel it into the `in_shardings` but I feel that this is not as clean: we would have to make fake devices when all we want is to say take the existing shardings and lowering logic and just pretend they are on a different platform. If we have multiple platforms then there is no clean way to put all the platforms into `in_shardings`.

Note that `jit` will not need to take the platform override parameters, only the lowering logic needs to.

(I should point out that I am already experimenting to verify that it is indeed possible to do multi-platform lowering in JAX without too much trouble and that the result is an HLO file that is only marginally larger than a single-platform HLO. I will be comparing with an alternative where we lower twice and we concatenate the resulting HLO.)

Reply:
For the more general purpose thing, I would like to see a design doc before we make changes to the API.

> PR will be changed to replace lowering_platform_override with lowering_platforms_override (not the plural).

Or maybe, we can repurpose the `backend` argument that jit already takes to do multi-platform lowering? I don't think we want to introduce arguments on `lower` that only work for some primitives and not for others.